mesonbuild/dependencies/base.py

```diff
@@ -1095,7 +1095,17 @@ def _get_cmake_info(self):
             return None
 
         # Extract the variables and sanity check them
+        root_paths = set(temp_parser.get_cmake_var('MESON_FIND_ROOT_PATH'))
+        root_paths.union(set(temp_parser.get_cmake_var('MESON_CMAKE_SYSROOT')))
+        root_paths = sorted(root_paths)
+        root_paths = list(filter(lambda x: os.path.isdir(x), root_paths))
         module_paths = sorted(set(temp_parser.get_cmake_var('MESON_PATHS_LIST')))
+        if len(root_paths) > 0:
+            unrooted_paths = module_paths
```

Comment:
Please make an actual copy here (`unrooted_paths = list(module_paths)`) or use `module_paths` directly, because lists are *mutable* in python. The current code can lead to hard to track bugs in the future, should `unrooted_paths` be modified, which would also modify `module_paths`.

Reply:
Nice catch :) I am an occational python developer, so this wasn't visible to me at all. Will fix that.

Reply:
I've changed the logic to append "rooted" (prepended with `CMAKE_SYSROOT` and `CMAKE_FIND_ROOT_PATH`) directories to `module_paths`.